book/crud.rst

```diff
@@ -0,0 +1,767 @@
+                ->add('boughtAt')
+                ->add('leasedUntil')
+            ;
+        }
+
+        public function setDefaultOptions(OptionsResolverInterface $resolver)
+        {
+            $resolver->setDefaults(array(
+                'data_class' => 'InventoryBundle\Entity\Vehicle',
+            ));
+        }
+
+        public function getName()
+        {
+            return 'inventory_vehicle';
+        }
+    }
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/InventoryBundle/Resources/config/form.yml
+    services:
+        inventory.form.type.vehicle:
+            class: InventoryBundle\Form\Type\VehicleType
+            tags:
+                - { name: form.type, alias: inventory_vehicle }
+
+Then, you will need to create the needed controller actions. You can simplify the actions if you
+create a dedicated method that is handling the form submission:
+
+.. code-block:: php
+    :linenos:
+
+    // src/InventoryBundle/Controller/VehicleController.php
+    namespace InventoryBundle\Controller;
+
+    // ...
+    use InventoryBundle\Entity\Vehicle;
+    use Symfony\Component\HttpFoundation\Request;
+
+    class VehicleController extends Controller
+    {
+        /**
+         * @Route("/create", name="inventory.vehicle_create")
+         * @Template("InventoryBundle:Vehicle:update.html.twig")
+         * @Acl(
+         *     id="inventory.vehicle_create",
+         *     type="entity",
+         *     class="InventoryBundle:Vehicle",
+         *     permission="CREATE"
+         * )
+         */
+        public function createAction(Request $request)
+        {
+            return $this->update(new Vehicle(), $request);
+        }
+
+        /**
+         * @Route("/update/{id}", name="inventory.vehicle_update", requirements={"id":"\d+"}, defaults={"id":0})
+         * @Template()
+         * @Acl(
+         *     id="inventory.vehicle_update",
+         *     type="entity",
+         *     class="InventoryBundle:Vehicle",
+         *     permission="EDIT"
+         * )
+         */
+        public function updateAction(Vehicle $vehicle, Request $request)
+        {
+            return $this->update($vehicle, $request);
+        }
+
+        private function update(Vehicle $vehicle, Request $request)
+        {
+            $form = $this->get('form.factory')->create('inventory_vehicle', $vehicle);
+            $form->handleRequest($request);
+
+            if ($form->isSubmitted() && $form->isValid()) {
+                $entityManager = $this->getDoctrine()->getEntityManager();
+                $entityManager->persist($vehicle);
+                $entityManager->flush();
+
+                return $this->get('oro_ui.router')->redirectAfterSave(
+                    array(
+                        'route' => 'inventory.vehicle_update',
+                        'parameters' => array('id' => $vehicle->getId()),
+                    )
+                    array('route' => 'inventory.vehicle_index')
```

Comment:
It's a good practice to pass the $entity as the third parameter as it needs to check for ACL permissions. If not passed, the "Save and Stay" functionality may not work